### ISSUE TITLE
Redit command deletes an appointment when details match another appointment

### ISSUE THREAD

**Issue by meerkatboy** [meerkatboy/pe#6]
As a user, I may make edits to my reminders in the event where changes are to be made. However, using the redit command, if I edit a reminder to match the same datetime as another appointment and having the same details, both appointments will be merged into one appointment. This effectively deletes an existing appointment.

after step 1:
![image.png](https://raw.githubusercontent.com/meerkatboy/pe/main/files/9bc2683d-fd53-4152-954d-d05d55f3b985.png)
after step 2:
![image.png](https://raw.githubusercontent.com/meerkatboy/pe/main/files/dad76d74-c636-4329-b429-9afcbb348727.png)
after step 3:
![image.png](https://raw.githubusercontent.com/meerkatboy/pe/main/files/7d1fb790-4a7c-458f-9f42-49f160417750.png)

Steps to replicate:
1. radd n/Ahmad Syuaib dt/2024-10-31 16:00 d/lunch
2. radd n/Ahmad Syuaib dt/2024-10-10 16:00 d/lunch
3. redit 1 dt/2024-10-31 16:00

This cancels out 1 appointment and merges them. This can be counter-productive if the user were to accidentally input the wrong date and time and it matches the date and time of another appointment. This will cause an appointment to be merged with another, and deleting it. If the user does not manually remember the original appointment date and time of the initial appointment, he/she will not be able to get it back, since there is also no undo function in this application. This can lead to important reminders and appointments not being shown, highlighting a crucial problem that may be faced. Furthermore, this is not highlighted in the User Guide.

Instead of merging both appointments, an error message can be shown if the user tries to edit a reminder such that the edited version clashes with another appointment.
